internal/devpkg/package.go

```diff
@@ -85,12 +89,16 @@ type Package struct {
 	normalizedPackageAttributePathCache string // memoized value from normalizedPackageAttributePath()
 }
 
-// PackageFromStrings constructs Package from the list of package names provided.
+// PackagesFromStrings constructs Package from the list of package names provided.
 // These names correspond to devbox packages from the devbox.json config.
-func PackageFromStrings(rawNames []string, l lock.Locker) []*Package {
+func PackagesFromStrings(
+	rawNames []string, l lock.Locker, disablePlugin bool,
```

Comment:
Eh, I'm not sure what the best path forward is:
1. Keep as-is. Its odd, but caller is forced to confront the `disablePlugin` param
2. Omit `disablePlugin` param, and only set it in the `devbox.addPackages` call by manually looping over the `[]*Packages`. But it risks future callers not considering it.

Reply:
Your second comment is precisely why I included it. We risk not doing it in other callsites. In general, adding the field to `devpkg.Package` is risky because of this, but IMO it's the best way. The unit tests in this PR are actually pretty robust. If we accidentally break this, I think they would catch most issues.

There's another approach which is safer, but a bit more involved:

Currently `plugin.Manager` accepts `devpkg.Package`. But I can change that for it to accept a new `plugin.Package` which wraps `devpkg.Package`. This way we remove the `DisablePlugin` logic from `devpkg.Package` altogether. I'll look into this.